assets/js/blocks/checkout/address-card/index.tsx

```diff
@@ -33,7 +34,7 @@ const AddressCard = ( {
 				<div className="wc-block-components-address-card__address-section">
 					{ [
 						address.address_1,
-						address.address_2,
+						! fieldConfig.address_2.hidden && address.address_2,
```

Comment:
I also went along and fixed address_2 so it's not visible if the field is not visible.